src/overlays/actors/ovl_Dm_Tsg/z_dm_tsg.c

```diff
@@ -28,14 +28,66 @@ ActorInit Dm_Tsg_InitVars = {
     /**/ DmTsg_Draw,
 };
 
-#endif
+void DmTsg_Init(Actor* thisx, PlayState* play) {
+    DmTsg* this = THIS;
+    s32 i;
 
-extern UNK_TYPE D_06011458;
+    if (gSaveContext.save.entrance == ENTRANCE(OPENING_DUNGEON, 0)) {
+        Actor_SetScale(&this->actor, 0.1f);
+        for (i = 0; i < 100; i++) {
+            this->unk_78C[i] = (Rand_ZeroOne() * 65535.0f);
+            this->unk_856[i] = (s16)(Rand_ZeroOne() * 200.0f) + 100;
+        }
+    } else {
+        Actor_Kill(&this->actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Tsg/DmTsg_Init.s")
+void DmTsg_Destroy(Actor* thisx, PlayState* play) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Tsg/DmTsg_Destroy.s")
+void DmTsg_Update(Actor* thisx, PlayState* play) {
+    DmTsg* this = THIS;
+    s32 cueChannel;
+    s16 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Tsg/DmTsg_Update.s")
+    for (i = 0; i < 100; i++) {
+        this->unk_5F8[i] = (Math_SinS((i & 0xF) * this->unk_854) * 60.0f) + 120.0f;
```

Comment:
Would this match?
```suggestion
        this->unk_5F8[i] = (Math_SinS((i % 16) * this->unk_854) * 60.0f) + 120.0f;
```

Reply:
It doesn't, already tried.

Reply:
What about with `% 16U`?

Reply:
I think & 15 is more clear than % 16U

Reply:
Why would doing a bitwise operation with a dec number be clearer than an hex number?
Also, in this same scope the code is operating `i` with 16, so using 16 is more consistent

Reply:
Yeah i didn't thought about that, i'll change it to % 16U although i don't see a dev writing that U in my mind for some reason.

Reply:
Take in mind this can be the result of a macro, a `sizeof` or something like that